File: twitterClient/twitterClient/src/stomp/frames/SendFrame.cpp
//
//  SendFrame.cpp
//  Object to represent the SEND frame type
//
//  Authors:    Ken Saggy
//              Lidan Hifi
//
//  Copyright 2013 Lidan Hifi, Ken Saggy

#include "../../../include/stomp/frames/SendFrame.h"

namespace stomp {
    SendFrame::SendFrame(std::string destination, std::string body): StompFrame(SEND_FRAME_NAME, body) {
        addHeaderPair("destination", destination);
    }
    
    void SendFrame::accept(FrameVisitor *visitor)
    { }
}

File: twitterEngine/src/spl/stomp/protocol/SendFrame.java
package spl.stomp.protocol;

/**
 * @name Send Frame
 * @author  Lidan Hifi
 * @author  Ken Saggy
 * 
 * SEND stomp frame
 */
class SendFrame extends StompFrame {
    public SendFrame() {
        super(StompCommand.SEND);
    }
    
    public SendFrame(String destination, String body) {
        super(StompCommand.SEND, body);
        addFrameHeader("destination", destination);
    }
    
    @Override
    protected String[] getMandatoryHeaders() {
        return new String[] {"destination"};
    }
}
